app/lib/search/index_simple.dart

```diff
@@ -231,7 +231,29 @@ class SimplePackageIndex implements PackageIndex {
         return Score.max([name, descr, readme]).removeLowValues(
             fraction: 0.01, minValue: 0.001);
       }).toList();
-      return Score.multiply(wordScores);
+      Score score = Score.multiply(wordScores);
+
+      // filter results based on exact phrases
+      final List<String> exactPhrases = extractExactPhrases(text);
+      if (exactPhrases.isNotEmpty) {
+        final Map<String, double> matched = <String, double>{};
+        for (String package in score.getKeys()) {
+          final doc = _packages[package];
+          bool matchedAllPhrases = true;
+          for (String phrase in exactPhrases) {
+            final bool matchedPhrase = matchesPhrase(doc.package, phrase) ||
+                matchesPhrase(doc.description, phrase) ||
+                matchesPhrase(doc.readme, phrase);
```

Comment:
Given the cost of this, I would prefer if we do not do this on the `readme` file.

Reply:
We limit the readme length before it even reaches the index: https://github.com/dart-lang/pub-dartlang-dart/blob/master/app/lib/search/backend.dart#L101